xonsh/readline_shell.py

```diff
@@ -67,16 +83,34 @@ def setup_readline():
 
     # handle tab completion differences found in libedit readline compatibility
     # as discussed at http://stackoverflow.com/a/7116997
-    if readline.__doc__ and 'libedit' in readline.__doc__:
+    if uses_libedit and ON_DARWIN:
         readline.parse_and_bind("bind ^I rl_complete")
+        print('\n'.join(['', "*"*78,
+            "libedit detected - readline will not be well behaved, including but not limited to:",
+            "   * crashes on tab completion",
+            "   * incorrect history navigation",
+            "   * corrupting long-lines",
+            "   * failure to wrap or indent lines properly",
+            "",
+            "It is highly recommended that you install gnureadline, which is installable with:",
+            "     pip install gnureadline",
+            "*"*78]))
     else:
         readline.parse_and_bind("tab: complete")
     # try to load custom user settings
-    try:
-        readline.read_init_file()
-    except Exception:
-        # this seems to fail with libedit
-        print_exception('xonsh: could not load readline default init file.')
+    inputrc_name = os.environ.get('INPUTRC')
+    if inputrc_name is None:
+        if uses_libedit:
+            inputrc_name = '.editrc'
+        else:
+            inputrc_name = '.inputrc'
+        inputrc_name = os.path.join(os.path.expanduser('~'), inputrc_name)
+    if os.path.isfile(inputrc_name):
```

Comment:
This will ignore the presense of a global file, `/etc/inputrc`,  https://www.gnu.org/software/bash/manual/html_node/Readline-Init-File.html  I think that if `inputrc_name` is not a file, then `inputrc_file = None`. This will allow the readline library to look up its suite of default locations.